packages/bee-q/src/components/notification/bq-notification.tsx

```diff
@@ -0,0 +1,188 @@
+import { h, Component, Prop, Element, State, Method, Host, Watch } from '@stencil/core';
+import { getColorCSSVariable, validatePropValue, isDefined } from '../../shared/utils';
+import { NOTIFICATION_TYPES, TNotificationType } from './bg-notification.types';
+
+/**
+ * @part base - The component's internal wrapper of the notification component.
+ * @part icon - `<div>` container element of notification icon component. Will be shown if Prop showIcon is true.
+ * @part title - `<div>` container element of notification default slot.
+ * @part avatar - `<div>` container element of notification avatar component slot.  Will be shown if Prop showIcon is false.
+ * @part description - `<div>` container element of notification description slot
+ * @part footer - `<div>` container element of notification footer slot
+ * @part close-icon - `<div>` container element of notification close icon component.  Will be shown if Prop showClose is false.
+ */
+@Component({
+  tag: 'bq-notification',
+  styleUrl: './scss/bq-notification.scss',
+  shadow: true,
+})
+export class BqNotification {
+  // Own Properties
+  // ====================
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqNotificationElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  @State() private isHidden = true;
```

Comment:
We could use a mutable `isOpen` property instead of state.

```tsx
/** If true, the notification will be shown */
@Prop({ reflect: true, mutable: true }) isOpen = false;
```